hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/util/AirdropHandlerHelper.java

```diff
@@ -181,7 +183,14 @@ public static NftAirdropLists separateNftTransfers(
     private static boolean isPendingAirdrop(@NonNull Account receiver, @Nullable TokenRelation tokenRelation) {
         // check if we have existing association or free auto associations slots or unlimited auto associations
         if (tokenRelation != null) {
+            if (receiver.receiverSigRequired()) {
+                throw new HandleException(RECEIVER_SIG_REQUIRED);
```

Comment:
Hmmm why do we need this. The use cases changed. 
If the receiverSigRequired =true
1. if receiver signed and if associated, it will just be a normal transfer
2. if receiver signed and If there are open slots, then also it will be a normal transfer and will be auto associated
3. If receiver signed and no open slots, will go to pending transfer
4. If receiver didn't sign, it will go to pending transfer

it is updated here https://github.com/hashgraph/hedera-improvement-proposal/pull/1025/files

Reply:
Reworked with the new requirements